src/libraries/System.Private.CoreLib/src/System/Text/EncoderLatin1BestFitFallback.cs

```diff
@@ -8,81 +8,40 @@
 
 using System.Diagnostics;
 using System.Globalization;
-using System.Threading;
 
 namespace System.Text
 {
-    internal class InternalEncoderBestFitFallback : EncoderFallback
+    internal sealed class EncoderLatin1BestFitFallback : EncoderFallback
     {
-        // Our variables
-        internal Encoding _encoding;
-        internal char[]? _arrayBestFit = null;
+        // Provides access to the singleton instance of this fallback mechanism
+        internal static readonly EncoderLatin1BestFitFallback SingletonInstance = new EncoderLatin1BestFitFallback();
 
-        internal InternalEncoderBestFitFallback(Encoding encoding)
+        private EncoderLatin1BestFitFallback()
         {
-            // Need to load our replacement characters table.
-            _encoding = encoding;
         }
 
         public override EncoderFallbackBuffer CreateFallbackBuffer() =>
-            new InternalEncoderBestFitFallbackBuffer(this);
+            new EncoderLatin1BestFitFallbackBuffer();
 
         // Maximum number of characters that this instance of this fallback could return
         public override int MaxCharCount => 1;
-
-        public override bool Equals(object? value) =>
-            value is InternalEncoderBestFitFallback that &&
-            _encoding.CodePage == that._encoding.CodePage;
-
-        public override int GetHashCode() => _encoding.CodePage;
     }
 
-    internal sealed class InternalEncoderBestFitFallbackBuffer : EncoderFallbackBuffer
+    internal sealed partial class EncoderLatin1BestFitFallbackBuffer : EncoderFallbackBuffer
     {
         // Our variables
         private char _cBestFit = '\0';
-        private readonly InternalEncoderBestFitFallback _oFallback;
+        private readonly char[] _arrayBestFit = s_arrayCharBestFit;
```

Comment:
This field can be deleted and its use replaced by direct use of `s_arrayCharBestFit` .